### PR TITLE
webui(dm): always redircet to index page

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve?

Issue Number: ref #3583

### What is changed and how it works?

as title

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
  * start a dm-master with ui and access some page like "/dashboard/xxxxx", you should got a redircet to /dashboard/

Code changes

Side effects

 - Possible performance regression

Related changes

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by Ehco1996** [comment]
could you please take another look for this pr ? thanks 🧡🧡🧡 

@D3Hunter  @okJiang

**Comment by Ehco1996** [comment]
could you please take another look for this pr ? thanks 🧡🧡🧡
/cc @D3Hunter @okJiang

File: dm/ui/server.go [hunk 1]
```diff
@@ -18,27 +18,49 @@ package ui
...
-	stripped, err := fs.Sub(WebUIAssets, "dist")
+	stripped, err := fs.Sub(WebUIAssets, buildPath)
 	if err != nil {
 		panic(err) // this should never happen
 	}
 	return http.FS(stripped)
 }
 
+// we need this to handel this case: user want to access /dashboard/source.html/ but webui is a single page app,
+// and it only can handel requests in index page, so we need to redirect to index page.
```
Comment (D3Hunter) [inline dm/ui/server.go:45]:
```suggestion
// we need this to handle this case: user want to access /dashboard/source.html/ but webui is a single page app,
// and it only can handle requests in index page, so we need to redirect to index page.
```

**Review by D3Hunter — APPROVED** [review]
rest LGTM

**Review by lance6716 — APPROVED** [review]

**Comment by Ehco1996** [comment]
ping license/cla 😷😷

**Pull request merged**